modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -2887,6 +2887,13 @@ private Predicate _getRelationshipObjectFieldPredicate(
 			objectDefinition.getObjectDefinitionId(),
 			titleObjectField.getName());
 
+		Column<?, ?> primaryKeyColumn = table.getColumn(
+			objectDefinition.getPKObjectFieldDBColumnName());
+
+		if (primaryKeyColumn == null) {
+			return null;
```

Comment:
Hey @linolaoj, it doesn't seem to be the right solution, if we do this, when the user searches for something in the entries, this column of the relationship in question will not be included in the search.

Reply:
@guilhermedcamacho is right. This way, if the title objectField of the parent Object be some column belonging to ObjectEntry table (some metadata column), when we are in the child Object portlet and search by this information of the parent, the search won't work.

I believe would be better focus the adjust [at this moment](https://github.com/liferay/liferay-portal/blob/ba4392c890dcfb099cafbf939f1e293f3b0564a8/modules/apps/object/object-service/src/main/java/com/liferay/object/internal/entry/util/ObjectEntrySearchUtil.java#L87-L89)

doing that:
```java
   if (primaryKeyColumn == null) {
      primaryKeyColumn = ObjectEntryTable.INSTANCE.objectEntryId;
   }
```

Because, if this table has no column that match with `objectDefinition.getPKObjectFieldDBColumnName()` it means the current table is actually the ObjectEntry table, so we have to take account the **objectEntryId** column as primaryKey